index.bs

```diff
@@ -1349,15 +1350,31 @@ Canvas Rendering Context {#canvas-rendering-context}
 XRPresentationContext {#xrpresentationcontext-interface}
 ---------------------
 
+When the {{HTMLCanvasElement/getContext()}} method of an {{HTMLCanvasElement}} |canvas| is to return a new object for the <var ignore>contextId</var> <code>present-xr</code>, the user agent must return an {{XRPresentationContext}} with {{XRPresentationContext/canvas}} set to |canvas|.
+
 <pre class="idl">
 [SecureContext, Exposed=Window] interface XRPresentationContext {
   readonly attribute HTMLCanvasElement canvas;
 };
 </pre>
 
-Each {{XRPresentationContext}} has an associated <dfn attribute for="XRPresentationContext">canvas</dfn> which is an {{HTMLCanvasElement}} set during creation.
+The <dfn attribute for="XRPresentationContext">canvas</dfn> attribute indicates the {{HTMLCanvasElement}} that created this context.
+
+Each {{XRPresentationContext}} has an <dfn for="XRPresentationContext">session</dfn> which is an {{XRSession}} initially set to <code>null</code>.
+
+<div class="algorithm" data-algorithm="presentation-context-session-update">
+
+When another algorithm indicates it will <dfn>update the XRPresentationContext session</dfn> to {{XRSession}} |session|, run the following steps:
+
+  1. Let |context| be the target {{XRPresentationContext}}.
+  1. Let |prevSession| be |context|'s [=XRPresentationContext/session=].
+  1. If |prevSession| is equal to |session| abort these steps.
+  1. If |prevSession| is not <code>null</code>:
+    1. If |prevSession|'s {{XRSession/renderState}}'s {{XRRenderState/outputContext}} is equal to |context| set |prevSession|'s {{XRSession/renderState}}'s {{XRRenderState/outputContext}} to <code>null</code>.
+  1. Set |context|'s [=XRPresentationContext/session=] to |session|.
```

Comment:
If removing the output context from |prevSession| does need to be queued, does this last step also need to be waited on?